src/config/config.cpp

```diff
@@ -0,0 +1,316 @@
+#include "config.h"
+
+// Helpers
+
+bool is_all_scalar(YAML::Node value) {
```

Comment:
wrap your helpers in a namespace, and do `using namespace config_helpers` or something.